lib/shyaml.py

```diff
@@ -0,0 +1,174 @@
+# vim: set encoding=utf-8 tabstop=4 softtabstop=4 shiftwidth=4 expandtab
+#########################################################################
+# Copyright 2016-       Martin Sinn                         m.sinn@gmx.de
+#########################################################################
+#  This file is part of SmartHomeNG
+#  https://github.com/smarthomeNG/smarthome
+#  http://knx-user-forum.de/
+#
+#  SmartHomeNG is free software: you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License as published by
+#  the Free Software Foundation, either version 3 of the License, or
+#  (at your option) any later version.
+#
+#  SmartHomeNG is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with SmartHomeNG. If not, see <http://www.gnu.org/licenses/>.
+#########################################################################
+
+import logging
+import yaml
+from collections import OrderedDict
+
+
+logger = logging.getLogger(__name__)
+
+# ==================================================================================
+#   Routines to handle yaml files
+#
+
+def yaml_load(filename, ordered=False):
+    """
+    Load contents of a yaml file into an dict/OrderedDict structure
+
+    :param filename: name of the yaml file to load
+    :return: Dict/OrderedDict structure or None, if an error occured
+    """
+
+    dict_type = 'dict'
+    if ordered:
+        dict_type = 'OrderedDict'
+    logger.info("Loading '{}' to '{}'".format(filename, dict_type))
+    y = None
+
+    try:
+        with open(filename, 'r') as stream:
+            sdata = stream.read()
+        sdata = sdata.replace('\n', '\n\n')
+        if ordered:
+            y = _ordered_load(sdata, yaml.SafeLoader)
+        else:
+            y = yaml.load(sdata, yaml.SafeLoader)
+    except Exception as e:
+        logger.error("YAML-file load error:  \n'%s'" % (e))
+
+    return y
+
+
+def yaml_save(filename, data):
+    """
+    Save contents of an OrderedDict structure to a yaml file
+
+    :param filename: name of the yaml file to save to
+    :param data: OrderedDict to save
+    """
+
+    ordered = (type(data).__name__ == 'OrderedDict')
+    dict_type = 'dict'
+    if ordered:
+        dict_type = 'OrderedDict'
+    logger.info("Saving '{}' to '{}'".format(dict_type, filename))
+    if ordered:
+        sdata = _ordered_dump(data, Dumper=yaml.SafeDumper, indent=4, width=256, allow_unicode=True, default_flow_style=False)
+    else:
+        sdata = yaml.dump(data, Dumper=yaml.SafeDumper, indent=4, width=256, allow_unicode=True, default_flow_style=False)
+    sdata = _format_yaml_dump( sdata )
+    with open(filename, 'w') as outfile:
+        outfile.write( sdata )

```

Comment:
Do we want to start supporting writing/duming configuration into files? I guess this will just remove the user comments and formatting from a configuration file completely.